File: schemas/cms-blog/drizzle/post_revisions.ts
// post_revisions: Versioned snapshots of post content for revision history.
// See README.md for full design rationale.
import { pgTable, uuid, text, integer, timestamp, unique, index } from "drizzle-orm/pg-core";
import { posts } from "./posts";

export const postRevisions = pgTable("post_revisions", {
  id: uuid("id").primaryKey().defaultRandom(),
  postId: uuid("post_id").notNull().references(() => posts.id, { onDelete: "cascade" }),
  revisionNumber: integer("revision_number").notNull(),
  title: text("title").notNull(),
  content: text("content"),
  excerpt: text("excerpt"),
  createdBy: uuid("created_by").notNull(), // FK → users.id (restrict delete)
  createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
}, (table) => [
  unique("uq_post_revisions_post_id_revision_number").on(table.postId, table.revisionNumber),
  index("idx_post_revisions_post_id_created_at").on(table.postId, table.createdAt),
]);

File: schemas/cms-blog/spacetimedb/post_revisions.rs
// post_revisions: Immutable content snapshots for post version history.
// See README.md for full design rationale.
use spacetimedb::Timestamp;

#[spacetimedb::table(name = post_revisions, public)]
pub struct PostRevision {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub post_id: String, // FK -> posts.id (cascade delete)
    pub revision_number: i32, // Composite unique: (post_id, revision_number)
    pub title: String,
    pub content: Option<String>,
    pub excerpt: Option<String>,
    pub created_by: String, // FK → users.id (restrict delete)
    pub created_at: Timestamp,
}
